apps/google-analytics-4/frontend/src/hooks/useKeyService.tsx

```diff
@@ -37,60 +39,71 @@ export default function useKeyService(props: Props): KeyServiceInfoType {
   const [serviceAccountKeyFileIsRequired, setServiceAccountKeyFileIsRequired] =
     useState<boolean>(false);
 
-  const sdk = useSDK<AppExtensionSDK>();
-
-  const onConfigure = useCallback(async () => {
-    const currentState = await sdk.app.getCurrentState();
+  const sdk = useSDK<any>();
 
-    if (!serviceAccountKeyFileIsValid) {
-      sdk.notifier.error('Invalid service account key file. See field error for details');
-      return false;
-    }
+  init((sdkLocation) => {
+    if (sdkLocation.location.is(locations.LOCATION_APP_CONFIG)) setIsConfigureScreen(true);
+    else setIsConfigureScreen(false);
+  });
 
-    if (serviceAccountKeyFileIsRequired && !newServiceAccountKeyId) {
-      sdk.notifier.error('A valid service account key file is required');
-      return false;
-    }
-
-    const newServiceKeyParameters = {
-      serviceAccountKey: newServiceAccountKey,
-      serviceAccountKeyId: newServiceAccountKeyId,
-    };
+  const onConfigure = useCallback(async () => {
+    if (isConfigureScreen) {
+      const currentState = await sdk.app.getCurrentState();
 
-    const newParameters = Object.assign(
-      {},
-      parameters,
-      omitBy(newServiceKeyParameters, (val) => val === null)
-    );
+      if (!serviceAccountKeyFileIsValid) {
+        sdk.notifier.error('Invalid service account key file. See field error for details');
+        return false;
+      }
 
-    setParameters(newParameters);
-    if (onSaveGoogleAccountDetails) onSaveGoogleAccountDetails();
-    setServiceAccountKeyFileIsRequired(false);
-    setServiceAccountKeyFile('');
+      if (serviceAccountKeyFileIsRequired && !newServiceAccountKeyId) {
+        sdk.notifier.error('A valid service account key file is required');
+        return false;
+      }
 
-    return {
-      parameters: newParameters,
-      targetState: currentState,
-    };
+      const newServiceKeyParameters = {
+        serviceAccountKey: newServiceAccountKey,
+        serviceAccountKeyId: newServiceAccountKeyId,
+      };
+
+      const newParameters = Object.assign(
+        {},
+        parameters,
+        omitBy(newServiceKeyParameters, (val) => val === null)
+      );
+
+      setParameters(newParameters);
+      if (onSaveGoogleAccountDetails) onSaveGoogleAccountDetails();
+      setServiceAccountKeyFileIsRequired(false);
+      setServiceAccountKeyFile('');
+
+      return {
+        parameters: newParameters,
+        targetState: currentState,
+      };
+    }
   }, [
-    sdk.app,
+    sdk?.app,
     sdk.notifier,
     serviceAccountKeyFileIsValid,
     serviceAccountKeyFileIsRequired,
     newServiceAccountKeyId,
     newServiceAccountKey,
     parameters,
     onSaveGoogleAccountDetails,
+    isConfigureScreen,
   ]);
 
   useEffect(() => {
-    sdk.app.onConfigure(() => onConfigure());
-  }, [sdk, onConfigure]);
+    if (isConfigureScreen) sdk.app.onConfigure(() => onConfigure());
```

Comment:
This is a bit of a smell and more evidence that the hook is the wrong place to organize all this code. cc @ryunsong-contentful 

I have a suggestion down below that we avoid this hook altogether in your PR, since we have targeted it to be refactored away.

Reply:
Sounds good - need two points of clarification @jsdalton 
1. When you say suggestion below is that in some PR somewhere else? I am unable to locate the suggestion.
2. By my PR do you mean the PR for mapping the account to properties pr?

Reply:
Great idea. I think this is a good call to just pull the parameters off the sdk and avoid using `theKeyService` given the future refactor. Will adjust that right now.